### PR TITLE
OperationalInsights - update readme files for cli extension update

### PR DESCRIPTION
**Pull request by dabenhamMic**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [v] PowerShell
      - [v] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by dabenhamMic** [comment]
Hi @kairu-ms , following our conversation on updating OperationalInsights CLI and PS components, 
I have added both readme.cli.md file, readme.az.md file, and updated readme.md file with the new azure-cli-extensions tag as described here: https://github.com/Azure/autorest.az/blob/9ed1aabd353385f8bdffbfa9f3892478111a029c/doc/how-to-author-readme-file.md
I'm not sure why is this PR failing on SDK azure-cli-extensions , will appreciate your help

File: specification/operationalinsights/resource-manager/readme.az.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+## AZ
+
+These settings apply only when `--az` is specified on the command line.
+
+For new Resource Provider. It is highly recommended to onboard Azure CLI extensions. There's no differences in terms of customer usage. 
+
+``` yaml $(az) && $(target-mode) != core
```
Comment (qiaozha) [inline specification/operationalinsights/resource-manager/readme.az.md:7]:
```suggestion
``` yaml $(az) && $(target-mode) != core
``` yaml $(az) && $(target-mode) != `core`
```

File: specification/operationalinsights/resource-manager/readme.az.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+az:
+    extensions: loganalytics
+    namespace: azure.mgmt.loganalytics
+    package-name: azure-mgmt-loganalytics
+az-output-folder: $(azure-cli-extension-folder)/src/loganalytics
+python-sdk-output-folder: "$(az-output-folder)/azext_loganalytics/vendored_sdks/loganalytics"
+# add additional configuration here specific for Azure CLI
+# refer to the faq.md for more details
+```
+``` yaml $(az) && $(target-mode) == core
```
Comment (qiaozha) [inline specification/operationalinsights/resource-manager/readme.az.md:17]:
```suggestion
``` yaml $(az) && $(target-mode) == core
``` yaml $(az) && $(target-mode) == `core`
```

File: specification/operationalinsights/resource-manager/readme.az.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+az:
+    extensions: loganalytics
+    namespace: azure.mgmt.loganalytics
+    package-name: azure-mgmt-loganalytics
+az-output-folder: $(azure-cli-extension-folder)/src/loganalytics
```
Comment (qiaozha) [inline specification/operationalinsights/resource-manager/readme.az.md:22]: az-output-folder: $(azure-cli-folder)/src/azure-cli/azure/cli/command_modules/loganalytics

Comment (qiaozha) [inline specification/operationalinsights/resource-manager/readme.az.md:22]: az-output-folder: $(azure-cli-folder)/src/azure-cli/azure/cli/command_modules/loganalytics

Comment (qiaozha) [inline specification/operationalinsights/resource-manager/readme.az.md:22]:
```suggestion
az-output-folder: $(azure-cli-extension-folder)/src/loganalytics
az-output-folder: $(azure-cli-folder)/src/azure-cli/azure/cli/command_modules/loganalytics
```

**Comment by qiaozha** [comment]
https://github.com/Azure/autorest.az/blob/master/doc/how-to-author-readme-file.md The document here is the accurate one.

File: specification/operationalinsights/resource-manager/readme.az.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
+## AZ
+
+These settings apply only when `--az` is specified on the command line.
+
+For new Resource Provider. It is highly recommended to onboard Azure CLI extensions. There's no differences in terms of customer usage. 
+
+``` yaml $(az) && $(target-mode) != core
```
Comment (qiaozha) [inline specification/operationalinsights/resource-manager/readme.az.md:7]:
```suggestion
```

File: specification/operationalinsights/resource-manager/readme.az.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+az:
+    extensions: loganalytics
+    namespace: azure.mgmt.loganalytics
+    package-name: azure-mgmt-loganalytics
+az-output-folder: $(azure-cli-extension-folder)/src/loganalytics
+python-sdk-output-folder: "$(az-output-folder)/azext_loganalytics/vendored_sdks/loganalytics"
+# add additional configuration here specific for Azure CLI
+# refer to the faq.md for more details
+```
+``` yaml $(az) && $(target-mode) == core
```
Comment (qiaozha) [inline specification/operationalinsights/resource-manager/readme.az.md:18]:
```suggestion
```

File: specification/operationalinsights/resource-manager/readme.az.md [hunk 6]
```diff
@@ -0,0 +1,27 @@
+## AZ
+
+These settings apply only when `--az` is specified on the command line.
+
+For new Resource Provider. It is highly recommended to onboard Azure CLI extensions. There's no differences in terms of customer usage. 
+
+``` yaml $(az) && $(target-mode) != `core`
```
Comment (qiaozha) [inline specification/operationalinsights/resource-manager/readme.az.md:7]:
```suggestion
``` yaml $(az) && $(target-mode) != 'core'
```

File: specification/operationalinsights/resource-manager/readme.az.md [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+az:
+    extensions: loganalytics
+    namespace: azure.mgmt.loganalytics
+    package-name: azure-mgmt-loganalytics
+az-output-folder: $(azure-cli-extension-folder)/src/loganalytics
+python-sdk-output-folder: "$(az-output-folder)/azext_loganalytics/vendored_sdks/loganalytics"
+# add additional configuration here specific for Azure CLI
+# refer to the faq.md for more details
+```
+``` yaml $(az) && $(target-mode) == `core`
```
Comment (qiaozha) [inline specification/operationalinsights/resource-manager/readme.az.md:17]:
```suggestion
``` yaml $(az) && $(target-mode) == 'core'
```

File: specification/operationalinsights/resource-manager/readme.az.md [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+python-sdk-output-folder: "$(az-output-folder)/azext_loganalytics/vendored_sdks/loganalytics"
+# add additional configuration here specific for Azure CLI
+# refer to the faq.md for more details
+```
+``` yaml $(az) && $(target-mode) == 'core'
+az:
+    extensions: loganalytics
+    namespace: azure.mgmt.loganalytics
+    package-name: azure-mgmt-loganalytics
+az-output-folder: $(azure-cli-extension-folder)/src/loganalytics
```
Comment (qiaozha) [inline specification/operationalinsights/resource-manager/readme.az.md:22]:
```suggestion
```

**Comment by dabenhamMic** [comment]
Hi @qiaozha , qq,  after adding:
  - repo: azure-cli-extensions
  - repo: azure-powershell
  to the readme.md file - will the cli extension and PS will be auto-generated after finishing this pr?

is there anything else missing here? can we merge this pr?

**Comment by qiaozha** [comment]
@dabenhamMic You can find the artifacts here https://github.com/Azure/azure-rest-api-specs/pull/13999#issuecomment-821956305 and have a try whether this artifacts meet your requirement.

**Comment by qiaozha** [comment]
> @dabenhamMic You can find the artifacts here [#13999 (comment)](https://github.com/Azure/azure-rest-api-specs/pull/13999#issuecomment-821956305) and have a try whether this artifacts meet your requirement.

The install instructions are inside each artifacts preview changes PR link. you can click it and find the respective install instructions.

**Comment by dabenhamMic** [comment]
@qiaozha  could you please approve and merge?

**Review by qiaozha — APPROVED** [review]
LGTM

**Pull request merged**